lib/modules/manager/gomod/artifacts.ts

```diff
@@ -275,23 +276,38 @@ export async function updateArtifacts({
         config.postUpdateOptions?.includes('gomodTidy1.17') === true ||
         config.postUpdateOptions?.includes('gomodTidyE') === true ||
         (config.updateType === 'major' && isImportPathUpdateRequired));
+
     if (isGoModTidyRequired) {
       args = 'mod tidy' + tidyOpts;
       logger.debug('go mod tidy command included');
       execCommands.push(`${cmd} ${args}`);
     }
 
+    const goWorkSumFileName = upath.join(goModDir, 'go.work.sum');
     if (useVendor) {
-      args = 'mod vendor';
-      logger.debug('go mod tidy command included');
-      execCommands.push(`${cmd} ${args}`);
-      if (isGoModTidyRequired) {
-        args = 'mod tidy' + tidyOpts;
-        logger.debug('go mod tidy command included');
+      // If go env GOWORK returns a non-empty path, check that it exists and if
+      // it does, then use go workspace vendoring.
+      const goWorkEnv = await exec(`${cmd} env GOWORK`, execOptions);
+      const goWorkFile = goWorkEnv?.stdout?.trim() || '';
```

Comment:
why not simply checking file exists? it's much faster.

Reply:
I started there, but it doesn't seem robust to when `go.work` isn't located directly near the `go.mod` that the manager is operating on.

Reply:
the `go.work` is a silbling or in parent folder? then use this

https://github.com/renovatebot/renovate/blob/8e6f89b0da7e722fe82c878c7afe35687a539d4b/lib/util/fs/index.ts#L136-L142